src/array/cuda/spmm.cu

```diff
@@ -282,6 +282,107 @@ void CusparseCsrmm2(
   if (valptr)
     device->FreeWorkspace(ctx, valptr);
 }
+
+/*! Cusparse implementation of SpMM on Csr format. */
+template <typename DType, typename IdType>
+void CusparseCsrmm2Hetero(
+    const DLContext& ctx,
+    const CSRMatrix& csr,
+    const DType* B_data, const DType* A_data,
+    DType* C_data,
+    int x_length,
+    cudaStream_t strm_id) {
+  // We use csrmm2 to perform following operation:
+  // C = A x B, where A is a sparse matrix in csr format, B is the dense matrix for node
+  // feature tensor. However, since cusparse only supports column-major, while our tensor
+  // is stored in row-major, the actual computation is:
+  // C = trans(A x trans(B)).
+  // Currently, we use cublasXgeam to implement transposition and allocate intermediate
+  // workspace memory for this.
+  const int m = csr.num_rows;
```

Comment:
Because things like csr.num_rows and csr.indices->shape[0] are int64_t, I think we should have some checks that we are not exceeding `std::numeric_limits<int>::max` before performing the conversion:
```
CHECK_GE(std::numeric_limits<int>::max, csr.num_rows);
CHECK_GE(std::numeric_limits<int>::max, csr.num_cols);
CHECK_GE(std::numeric_limits<int>::max, csr.indices->shape[0]);
```